recipes/criterion/all/conanfile.py

```diff
@@ -0,0 +1,82 @@
+from os import path
+from conan import ConanFile
+from conan.tools.files import get, copy, rmdir, replace_in_file, rename
+from conan.tools.gnu import PkgConfigDeps
+from conan.tools.layout import basic_layout
+from conan.tools.meson import MesonToolchain, Meson
+
+class CriterionConan(ConanFile):
+    name = "criterion"
+    version = "2.4.1"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/Snaipe/Criterion"
+    topics = ("testing")
+    description = "A cross-platform C and C++ unit testing framework for the 21st century"
+
+    # Binary configuration
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    requires = ("libgit2/[>=1.0]", "libffi/[~3.4]")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            self.options.rm_safe("fPIC")
+
+    def configure(self):        
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def generate(self):
+        pc = PkgConfigDeps(self)
+        pc.generate()
+
+        tc = MesonToolchain(self)
+        tc.project_options['force_fallback_for'] = ['boxfort']
+        tc.project_options['default_library'] = 'shared' if self.options['shared'] else 'static'
+        
+        # We don't need tests or samples
+        tc.project_options['tests'] = False
+        tc.project_options['samples'] = False
+        
+        tc.generate()
+
+    def _download_sources(self):
```

Comment:
This is the responsibility of the `source()` method, any reason not to use that?

Reply:
I based the recipe in this existing one: https://github.com/conan-io/conan-center-index/blob/master/recipes/approvaltests.cpp/all/conanfile.py where they just skip the source method all together. If I add more things, like the required submodules, I will get a check failure.